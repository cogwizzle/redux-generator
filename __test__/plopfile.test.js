jest.mock('../scripts/utils/loadUtils', () => ({
  loadSettings: jest
    .fn()
    .mockReturnValueOnce({ isSass: true })
    .mockReturnValueOnce({ isSass: false }),
  applySettings: jest.fn(),
}))
const use = require('../plopfile.js')

it('plop', () => {
  const plopMock = {
    setGenerator: jest.fn(),
  }

  use(plopMock)
  expect(plopMock.setGenerator).toHaveBeenCalledTimes(1)
})

it('plop action', () => {
  let expectedResults = null
  const plopMock = {
    setGenerator: jest.fn((name, description) => {
      expectedResults = description.actions({ name: 'test' })
    }),
  }

  use(plopMock)
  expect(expectedResults).toMatchSnapshot()
})

