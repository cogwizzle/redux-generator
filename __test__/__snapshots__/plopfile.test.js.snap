// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plop action 1`] = `
Array [
  Object {
    "path": "/home/jfehrman/redux-generator/src/redux/reducer/{{snakeCase name}}.undefined",
    "skipIfExists": false,
    "template": undefined,
    "type": "add",
  },
  Object {
    "path": "/home/jfehrman/redux-generator/src/redux/creator/{{snakeCase name}}.undefined",
    "skipIfExists": false,
    "template": "",
    "type": "add",
  },
  Object {
    "path": "/home/jfehrman/redux-generator/src/redux/actions/{{snakeCase name}}.undefined",
    "skipIfExists": false,
    "template": "",
    "type": "add",
  },
  Object {
    "path": "/home/jfehrman/redux-generator/src/redux/reducer/{{snakeCase name}}.undefined",
    "pattern": /;\\\\n/g,
    "template": "
",
    "type": "modify",
  },
]
`;
